test/unit/math/test_ad.hpp

```diff
@@ -1542,22 +1732,26 @@ template <typename F, typename EigMat>
 void expect_ad_matvar(F&& f, const EigMat& x) {
   using stan::plain_type_t;
   using stan::math::promote_scalar_t;
-  using stan::math::sum;
   using stan::math::var;
   using stan::math::var_value;
+  using stan::math::test::type_name;
   using mat_var = promote_scalar_t<var, EigMat>;
+  using var_mat = var_value<plain_type_t<EigMat>>;
   mat_var A_mv = x;
   plain_type_t<decltype(f(A_mv))> A_mv_f;
-  using var_mat = var_value<plain_type_t<EigMat>>;
   var_mat A_vm = x;
   plain_type_t<decltype(f(A_vm))> A_vm_f;
+  // Check return type is correct
+  check_return_type(A_mv_f, A_mv);
+  check_return_type(A_vm_f, A_vm);
   // If one throws, the other should throw as well
   try {
     A_mv_f = f(A_mv);
   } catch (...) {
     try {
       f(A_vm);
-      FAIL() << "matrix<var> throws, expect var<matrix> version to throw";
+      FAIL() << type_name<mat_var>() << " throws, expected "
+             << type_name<var_mat>() << " version to throw";
     } catch (...) {
       SUCCEED();
```

Comment:
If both functions have thrown, you shouldn't go on to check the gradient or sum of the gradients.

Reply:
Ah good catch I though `SUCCEED()` stopped the test but I need to add a return; here to end early